staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -355,14 +355,66 @@ type ListOptions struct {
+	//
+	// watch semantics:
+	// - if unset, start a watch at the most recent resource version, which must be consistent
+	//   (i.e. served from etcd via a quorum read). To establish initial state, the watch begins
+	//   with synthetic “Added” events of all resources instances that exist at the starting
+	//   resource version. All following watch events are for all changes that occurred after
+	//   the resource version the watch started at;
+	// - if 0, start a watch at any resource version, the most recent resource version available
+	//   is preferred, but not required; any starting resource version is allowed. It is
+	//   possible for the watch to start at a much older resource version that the client
+	//   has previously observed, particularly in high availability configurations, due to
+	//   partitions or stale caches. Clients that cannot tolerate this should not start a
+	//   watch with this semantic. To establish initial state, the watch begins with synthetic
+	//   “Added” events for all resources instances that exist at the starting resource version.
+	//   All following watch events are for all changes that occurred after the resource version
+	//   the watch started at;
+	// - if non-zero, start a watch at an exact resource version. The watch events are for all
+	//   changes after the provided resource version. Watch is not started with synthetic “Added”
+	//   events for the provided resource version. The client is assumed to already have the
+	//   initial state at the starting resource version since the client provided the resource version.
+	//
+	// list semantics:
+	// - if unset, then the result is returned at the most recent resource version. The returned
+	//   data must be consistent (i.e. served from etcd via a quorum read);
+	// - if set and ResourceVersionMatch is set, ResourceVersion is applied according to the ResourceVersionMatch rule;
+	// - if set and ResourceVersionMatch is unset or the server ignores ResourceVersionMatch, the legacy behavior applies:
+	//   - if 0, the result may be at any resource version. The newest available resource version
+	//     is preferred, but strong consistency is not required; data at any resource version may
+	//     be served. It is possible for the request to return data at a much older resource version
+	//     that the client has previously observed, particularly in high availability configurations,
+	//     due to partitions or stale caches. Clients that cannot tolerate this should not use this
+	//     semantic;
+	//   - if non-zero and Limit is unset, the ResourceVersionMatchNotOlderThan rule applies implicitly;
+	//   - if non-zero and Limit is set, the ResourceVersionMatchExact rule applies implicitly.
+	//
+	// Defaults to unset
 	// +optional
 	ResourceVersion string `json:"resourceVersion,omitempty" protobuf:"bytes,4,opt,name=resourceVersion"`
+
+	// ResourceVersionMatch determines how ResourceVersion is applied to list calls.
```

Comment:
Thanks. I've adjusted everything to make sense in openapi doc.

Reply:
And since I've replaced much of the documentation here with a url. I'm keeping track of these edits in https://docs.google.com/document/d/1P228Uo6_-U5-jIgJG9m9X-DMocOgQCyf1ltC5WFekAE/edit